python_modules/dagster/dagster/core/definitions/dependency.py

```diff
@@ -637,6 +638,10 @@ def __new__(
         description: Optional[str] = None,
         node: Optional[str] = None,
     ):
+        if solid:
```

Comment:
We should use `canonicalize_backcompat_args` in `dagster.utils.backcompat` instead, it serves exactly this purpose and will keep error messages consistent.